chia/full_node/mempool_check_conditions.py

```diff
@@ -156,6 +158,45 @@ def get_spends_for_block(generator: BlockGenerator) -> List[CoinSpend]:
 
     return spends
 
+def get_spends_for_block_with_conditions(generator: BlockGenerator, height: int, constants: ConsensusConstants) -> List[Dict[str, Any]]:
```

Comment:
I'm not super excited about the type-unsafety bleed into this file. If at all possible, I think it would be preferable to return a `SpendBundleConditions` object. It won't be a verbatim list of all conditions though, ones that are "shadowed" by others will be dropped, and coin/puzzle announcements and assertions will too.